### PR TITLE
pathWhitelist: can I make a recursive inclusion?

### PR DESCRIPTION
**Pull request by jonboulle**
It's not clear if I can say "/foo and everything under it" or not.

### CODE REVIEW

**Comment by jonboulle** [comment]
ptal

File: SPEC.md [hunk 1]
```diff
@@ -765,7 +765,7 @@ JSON Schema for the Image Manifest (app image manifest, ACI manifest), conformin
...
+* **pathWhitelist** (list of strings, optional) complete whitelist of paths that may exist in the rootfs after assembly (i.e. unpacking the files in this image and overlaying its dependencies, in order). Paths must be absolute and are not recursive. If a path ends in a slash, then the ACE must ensure that the directory is empty - i.e., if the image or a dependency contains files beneath this path, they will not be present in the rendered filesystem. This field is only required if the app has dependencies and you wish to remove files from the rootfs before running the app. An empty list is equivalent to an absent value and means that all files in this image and any dependencies will be available in the rootfs.
```
Comment (thockin) [inline SPEC.md]: One more nit: "If a path ends in a slash" - what happens if I list a path that is a directory but I do not put a slash?  You already said dirs are not recursive.  Maybe this whole clause is moot?

**Comment by jonboulle** [comment]
@thockin Took this in a different direction.

File: SPEC.md [hunk 2]
```diff
@@ -168,8 +168,16 @@ This UUID is exposed to the pod through the [Metadata Service](#app-container-me
...
+Each app in a pod will start chrooted into its own unique read-write filesystem before execution.
+
+An app's filesystem must be *rendered* in an empty directory by the following process (or equivalent):
+- The `rootfs` contained in the ACI is extracted
+- If the ACI contains a non-empty `dependencies` field in its `ImageManifest`, the `rootfs` of each dependent image is extracted, in the order in which they are listed
+- If the ACI contains a non-empty `pathWhitelist` field in its `ImageManifest`, *all* paths not in the whitelist must be removed
+
+Every execution of an app MUST start from a clean copy of this rendered filesystem.
+
+The simplest implementation will take an ACI (gzip-compressed, with no dependencies) and extract it into a new directory:
```
Comment (thockin) [inline SPEC.md]: the gzip bit adds no value

**Comment by thockin** [comment]
LGTM

File: SPEC.md (hunk 2 above)
Comment (jonboulle) [inline SPEC.md]: eh, I added that in an earlier iteration where it made more sense, will remove.

**Pull request merged**